### PR TITLE
mattermost: 7.5.1 -> 7.5.2

### PR DESCRIPTION
**Pull request by numinit**
###### Description of changes

Mattermost v7.5.2 contains low to medium severity level security fixes. [Upgrading](https://docs.mattermost.com/upgrade/upgrading-mattermost-server.html) to this release is recommended. Details will be posted on our [security updates page](https://mattermost.com/security-updates/) 30 days after release as per the [Mattermost Responsible Disclosure Policy](https://mattermost.com/security-vulnerability-report/).

https://docs.mattermost.com/install/self-managed-changelog.html#release-v7-5-feature-release

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/mattermost/default.nix [hunk 1]
```diff
@@ -29,6 +30,11 @@ buildGoModule rec {
     "-s"
     "-w"
     "-X github.com/mattermost/mattermost-server/v6/model.Version=${version}"
+    "-X github.com/mattermost/mattermost-server/v6/model.BuildNumber=${version}-nixpkgs"
+    "-X github.com/mattermost/mattermost-server/v6/model.BuildDate=1970-01-01"
+    "-X github.com/mattermost/mattermost-server/v6/model.BuildHash=${gitRev}"
+    "-X github.com/mattermost/mattermost-server/v6/model.BuildHashEnterprise=${gitRev}"
+    "-X github.com/mattermost/mattermost-server/v6/model.BuildEnterpriseReady=false"
```
Comment (numinit) [inline pkgs/servers/mattermost/default.nix:37]: All the build info is empty in NixOS 22.11, this fixes it:

```
$ ./result/bin/mattermost version
Version: 7.5.2
Build Number: 7.5.2-nixpkgs
Build Date: 1970-01-01
Build Hash: v7.5.2
Build Enterprise Ready: false
```

**Comment by Kranzes** [comment]
Why does it not go into master directly?

File: pkgs/servers/mattermost/default.nix [hunk 2]
```diff
@@ -7,18 +7,19 @@
 
 buildGoModule rec {
   pname = "mattermost";
-  version = "7.5.1";
+  version = "7.5.2";
+  gitRev = "v${version}";
```
Comment (SuperSandro2000) [inline pkgs/servers/mattermost/default.nix:11]:
```suggestion
```

File: pkgs/servers/mattermost/default.nix [hunk 3]
```diff
@@ -7,18 +7,19 @@
...
 
   src = fetchFromGitHub {
     owner = "mattermost";
     repo = "mattermost-server";
-    rev = "v${version}";
-    sha256 = "sha256-uaud5yCJTV3/+5eqHvIurxM1EPtetTWeFTjkPKYmFiA=";
+    rev = gitRev;
```
Comment (SuperSandro2000) [inline pkgs/servers/mattermost/default.nix:16]:
```suggestion
    rev = "v${version}";
```

File: pkgs/servers/mattermost/default.nix [hunk 4]
```diff
@@ -29,6 +30,11 @@ buildGoModule rec {
     "-s"
     "-w"
     "-X github.com/mattermost/mattermost-server/v6/model.Version=${version}"
+    "-X github.com/mattermost/mattermost-server/v6/model.BuildNumber=${version}-nixpkgs"
+    "-X github.com/mattermost/mattermost-server/v6/model.BuildDate=1970-01-01"
+    "-X github.com/mattermost/mattermost-server/v6/model.BuildHash=${gitRev}"
```
Comment (SuperSandro2000) [inline pkgs/servers/mattermost/default.nix:35]:
```suggestion
    "-X github.com/mattermost/mattermost-server/v6/model.BuildHash=v${version}"
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by numinit** [comment]
> Why does it not go into master directly?

Per the [staging workflow](https://github.com/NixOS/rfcs/blob/master/rfcs/0026-staging-workflow.md) it appears this is where security fixes go since this is a stabilization fix that will be backported to 22.11.

File: pkgs/servers/mattermost/default.nix [hunk 5]
```diff
@@ -29,6 +29,11 @@ buildGoModule rec {
     "-s"
     "-w"
     "-X github.com/mattermost/mattermost-server/v6/model.Version=${version}"
+    "-X github.com/mattermost/mattermost-server/v6/model.BuildNumber=${version}-nixpkgs"
+    "-X github.com/mattermost/mattermost-server/v6/model.BuildDate=1970-01-01"
+    "-X github.com/mattermost/mattermost-server/v6/model.BuildHash=v${gitRev}"
+    "-X github.com/mattermost/mattermost-server/v6/model.BuildHashEnterprise=v${gitRev}"
```
Comment (Artturin) [inline pkgs/servers/mattermost/default.nix:35]: gitRev isn't set

**Comment by Artturin** [comment]
> > Why does it not go into master directly?
> 
> Per the [staging workflow](https://github.com/NixOS/rfcs/blob/master/rfcs/0026-staging-workflow.md?rgh-link-date=2022-12-29T20%3A44%3A53Z) it appears this is where security fixes go since this is a stabilization fix that will be backported to 22.11.

less than 500 rebuilds can go in to master

**Comment by SuperSandro2000** [comment]
> Per the [staging workflow](https://github.com/NixOS/rfcs/blob/master/rfcs/0026-staging-workflow.md?rgh-link-date=2022-12-29T20%3A44%3A53Z) it appears this is where security fixes go since this is a stabilization fix that will be backported to 22.11.

Only mass rebuilds go to staging and mass rebuild **with** security fixes go to staging-next.

**Comment by numinit** [comment]
Ah, good to know. Thanks, I'll rebase.

**Review by LeSuisse — APPROVED** [review]
nixpkgs-review happy on x86_64, test succeeds and no surprise in the changelog.

Result of `nixpkgs-review pr 208185` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mattermost</li>
  </ul>
</details>

**Pull request merged**